go/ql/test/library-tests/semmle/go/frameworks/Fasthttp/fasthttp.go

```diff
@@ -0,0 +1,206 @@
+		requestCtx.URI().QueryArgs().Peek("arg1")                   // $ UntrustedFlowSource="call to Peek"
+		requestCtx.URI().QueryArgs().PeekBytes([]byte("arg1"))      // $ UntrustedFlowSource="call to PeekBytes"
+		requestCtx.URI().QueryArgs().PeekMulti("arg1")              // $ UntrustedFlowSource="call to PeekMulti"
+		requestCtx.URI().QueryArgs().PeekMultiBytes([]byte("arg1")) // $ UntrustedFlowSource="call to PeekMultiBytes"
+		requestCtx.URI().QueryArgs().QueryString()                  // $ UntrustedFlowSource="call to QueryString"
+		requestCtx.URI().QueryArgs().String()                       // $ UntrustedFlowSource="call to String"
+		requestCtx.String()                                         // $ UntrustedFlowSource="call to String"
+		// not sure what is the best way to write query for following
+		//requestCtx.URI().QueryArgs().VisitAll(type func(,))
+
+		requestCtx.Path() // $ UntrustedFlowSource="call to Path"
+		// multipart.Form is already implemented
+		// requestCtx.FormFile("FileName")
+		// requestCtx.FormValue("ValueName")
+		requestCtx.Referer()           // $ UntrustedFlowSource="call to Referer"
+		requestCtx.PostBody()          // $ UntrustedFlowSource="call to PostBody"
+		requestCtx.RequestBodyStream() // $ UntrustedFlowSource="call to RequestBodyStream"
+		requestCtx.RequestURI()        // $ UntrustedFlowSource="call to RequestURI"
+		requestCtx.UserAgent()         // $ UntrustedFlowSource="call to UserAgent"
+		requestCtx.Host()              // $ UntrustedFlowSource="call to Host"
+
+		requestCtx.Request.Host()             // $ UntrustedFlowSource="call to Host"
+		requestCtx.Request.Body()             // $ UntrustedFlowSource="call to Body"
+		requestCtx.Request.RequestURI()       // $ UntrustedFlowSource="call to RequestURI"
+		requestCtx.Request.BodyGunzip()       // $ UntrustedFlowSource="call to BodyGunzip"
+		requestCtx.Request.BodyInflate()      // $ UntrustedFlowSource="call to BodyInflate"
+		requestCtx.Request.BodyUnbrotli()     // $ UntrustedFlowSource="call to BodyUnbrotli"
+		requestCtx.Request.BodyStream()       // $ UntrustedFlowSource="call to BodyStream"
+		requestCtx.Request.BodyUncompressed() // $ UntrustedFlowSource="call to BodyUncompressed"
+		requestCtx.Request.ReadBody(dstReader, 100, 1000)
+		requestCtx.Request.ReadLimitBody(dstReader, 100)
+		requestCtx.Request.ContinueReadBodyStream(dstReader, 100, true)
+		requestCtx.Request.ContinueReadBody(dstReader, 100)
+		// not sure what is the best way to write query for following
+		//requestCtx.Request.Header.VisitAllCookie()
+
+		// Response methods
+		// Xss Sinks Related method
+		userInput := "user Controlled input"
+		userInputByte := []byte("user Controlled input")
+		requestCtx.Response.AppendBody(userInputByte)   // $ XssSink=userInputByte
+		requestCtx.Response.AppendBodyString(userInput) // $ XssSink=userInput
+		rspWriter := requestCtx.Response.BodyWriter()   // IDK how to handle this that returns a `io.Writer`
```

Comment:
Comments like this should be in the models, not in the tests, because people won't read the tests as thoroughly. I will consult with colleagues on how best to model this.

Reply:
I have now made a suggestion in another comment that makes this test pass.